### ISSUE TITLE
ZFS master and 2.1.5 fails to boot root ZFS pool -can not import pool, name already exists

### ISSUE THREAD

**Issue by greg-hydrogen** [openzfs/zfs#13604]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Fedora
Distribution Version	| 35
Kernel Version	| 5.18.6
Architecture	| x86_64
OpenZFS Version	| current master and 2.1.5

### Describe the problem you're observing
After building the latest version from master and even 2.1.5 on reboot of the system, the root pool fails to mount staing
Can not import rpool as a pool with that name already exists

There is no pool that matches that name and rebooting into an older kernel with an older version of ZFS doesn't  have this issue

kernel 5.17.11-200.fc35.x86_64
zfs-2.1.99-1273_gc0bf952c8
zfs-kmod-2.1.99-946_g6b444cb97

I have upgraded this as I usually have and I believe it might be the recent dracut changes that has cause this to fail as I can see the pools but I get dropped into an emergency shell

Currently my rootfs is setup with the bootfs attribute, no additional kernel parameters and the zfs mountpoint set to /
### Describe how to reproduce the problem
install any version of zfs that is about 2 weeks older than the build above and it fails

**Comment by sysedwinistrator** [comment]
Hi,
I experienced the same issue on Arch Linux (using dracut for initramfs) after upgrading to 2.15. It looks like the ZFS dracut module always installs and activates the `zfs-import-scan.service` or `zfs-import-cache.service` systemd units (even if they're both disabled on the host that builds the initramfs). But then the initramfs additionally starts a script which tries to import the pool that has already been by a systemd service imported and fails.

As a workaround I commented out those lines in the ZFS module:
```sh
diff --git a/contrib/dracut/90zfs/mount-zfs.sh.in b/usr/lib/dracut/modules.d/90zfs/mount-zfs.sh
index fa9f1bb76..6a7595bad 100755
--- a/contrib/dracut/90zfs/mount-zfs.sh.in
+++ b/usr/lib/dracut/modules.d/90zfs/mount-zfs.sh
@@ -82,15 +82,15 @@ ZFS_DATASET="${ZFS_DATASET:-${root}}"
 ZFS_POOL="${ZFS_DATASET%%/*}"
 
 
-if ! zpool get -Ho name "${ZFS_POOL}" > /dev/null 2>&1; then
-    info "ZFS: Importing pool ${ZFS_POOL}..."
-    # shellcheck disable=SC2086
-    if ! zpool import -N ${ZPOOL_IMPORT_OPTS} "${ZFS_POOL}"; then
-        warn "ZFS: Unable to import pool ${ZFS_POOL}"
-        rootok=0
-        return 1
-    fi
-fi
+#if ! zpool get -Ho name "${ZFS_POOL}" > /dev/null 2>&1; then
+#    info "ZFS: Importing pool ${ZFS_POOL}..."
+#    # shellcheck disable=SC2086
+#    if ! zpool import -N ${ZPOOL_IMPORT_OPTS} "${ZFS_POOL}"; then
+#        warn "ZFS: Unable to import pool ${ZFS_POOL}"
+#        rootok=0
+#        return 1
+#    fi
+#fi
 
 # Load keys if we can or if we need to
 # TODO: for_relevant_root_children like in zfs-load-key.sh.in
``` 

That section was introduced by eaf1e060453ad6a335b708c5e724092741d6d1d3. I tested the first command, which tests whether the pool of the given name is already imported, and it doesn't work.  
`zpool get -Ho name tank` returns:
`bad property list: invalid property 'tank'`
But `zpool get -H name tank` works:
`tank	name	tank	-`
Since we only care about the exit code, we could use this instead.
But I don't know whether the script never worked or whether this is a bug in the zpool command.

**Comment by behlendorf** [comment]
Duplicate of #13600.  A fix for this was applied in #13602, we'll get it in the next 2,1 point release.  For now you'll have to manually tweak the script, sorry about the inconvenience!
